internal/input/input.go

```diff
@@ -0,0 +1,149 @@
+package input
```

Comment:
This was a great one to pull out. I like the separation of concerns here.

Reply:
thanks